packages/components/inputs/select-utils/src/create-select-styles.ts

```diff
@@ -43,25 +44,35 @@ const controlStyles = (props: TProps) => (base: TBase, state: TState) => {
   return {
     ...base,
     fontSize: designTokens.fontSizeForInput,
-    backgroundColor: props.isDisabled
-      ? designTokens.backgroundColorForInputWhenDisabled
-      : designTokens.backgroundColorForInput,
+    backgroundColor: (() => {
+      if (props.isDisabled)
+        return designTokens.backgroundColorForInputWhenDisabled;
+      if (props.isReadOnly)
+        return designTokens.backgroundColorForInputWhenReadonly;
+      return designTokens.backgroundColorForInput;
+    })(),
     borderColor: (() => {
       if (props.isDisabled) return designTokens.borderColorForInputWhenDisabled;
       if (state.isFocused) return designTokens.borderColorForInputWhenFocused;
       if (props.hasError) return designTokens.borderColorForInputWhenError;
       if (props.hasWarning) return designTokens.borderColorForInputWhenWarning;
-      if (props.isReadOnly) return designTokens.borderColorForInputWhenReadonly;
+      if (props.isReadOnly)
+        return designTokens.borderColorForSelectInputWhenReadonly;
       return designTokens.borderColorForInput;
     })(),
+    borderWidth: (() => {
+      if (props.hasError) return designTokens.borderWidthForSelectInput;
+      if (props.hasWarning) return designTokens.borderWidthForSelectInput;
```

Comment:
It seems both conditions use the same value:
```suggestion
      if (props.hasError || props.hasWarning) return designTokens.borderWidthForSelectInput;
```